tests/tools/test_parmetis_preproc.py

```diff
@@ -0,0 +1,236 @@
+    edges["n1:e1:n1"]["format"]["delimiter"] = edge_fmt_del
+
+    os.makedirs(edges_dir, exist_ok=True)
+    fmt_meta = {"name": edge_fmt}
+    if edge_fmt == "csv":
+        fmt_meta["delimiter"] = edge_fmt_del
+
+    for idx in range(num_chunks):
+        path = os.path.join(edges_dir, f"test_edges_{idx}")
+        array_parser = array_readwriter.get_array_parser(**fmt_meta)
+        edge_data = (
+            np.array([np.arange(10), np.arange(10)]).reshape(10, 2) + 10 * idx
+        )
+        array_parser.write(path, edge_data)
+
+    edge_files = [path]
+    edges["n1:e1:n1"]["data"] = edge_files
+    schema["edges"] = edges
+
+    return schema
+
+
+@pytest.mark.parametrize("num_chunks, num_parts", [[4, 1], [4, 2], [4, 4]])
+@pytest.mark.parametrize("edges_fmt", ["csv", "parquet"])
+def test_gen_edge_files(num_chunks, num_parts, edges_fmt):
+    """Unit test case for the function
+    tools/distpartitioning/parmetis_preprocess.py::gen_edge_files
+
+    Parameters:
+    -----------
+    num_chunks : int
+        no. of chunks the input graph needs to be partititioned into
+    num_parts : int
+        no. of partitions
+    edges_fmt : string
+        specifying the storage format for the edge files
+    """
+
+    # Create the input dataset
+    with tempfile.TemporaryDirectory() as root_dir:
+
+        # Prepare the state information for firing unit test
+        input_dir = os.path.join(root_dir, "chunked-data")
+        output_dir = os.path.join(root_dir, "preproc_dir")
+
+        # Get the parser object
+        fn_params = namedtuple("fn_params", "input_dir output_dir num_parts")
+        fn_params.input_dir = input_dir
+        fn_params.output_dir = output_dir
+        fn_params.num_parts = num_parts
+
+        # Read the input schema
+        schema_map = _get_test_data(input_dir, num_chunks)
+
+        # Get the global node id offsets for each node type
+        # There is only one node-type in the test graph
+        # which range from 0 thru 9.
+        ntype_gnid_offset = {}
+        ntype_gnid_offset["n1"] = np.array([0, 10 * num_chunks]).reshape(1, 2)
+
+        # Iterate over no. of partitions
+        for rank in range(num_parts):
+
+            # Fire the unit test case and get the results
+            actual_results = gen_edge_files(rank, schema_map, fn_params)
+
+            # Get the gold results for baseline comparision, expected results
+            baseline_results, fmt_results, edge_types = _get_edge_files(
+                schema_map, rank, num_parts
+            )
+
+            # Validate the results with the baseline results
+            # Test 1. no. of files should have the same count per rank
+            assert len(baseline_results) == len(actual_results)
+
+            # Test 2. Check the contents of each file and verify the
+            # file contents
+            for idx, fname in enumerate(baseline_results):
+
+                # Check the file exists
+                assert os.path.isfile(fname)
```

Comment:
Is this testing that `_get_edge_files` works? Assertion doesn't have to do with `gen_edge_files` functionality

Reply:
Now this unit test iterates over the results from the 'gen_edge_files' function to address the above concern